### PR TITLE
BUG: Add basic __format__ for masked element to fix incorrect print

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
This PR adds one commit to gh-15410.  The two changes in the new commit were suggested by @seberg in comments to that PR.

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
I rebased on master to get the latest changes to the CI infrastructure.  The Azure Linux PyPy3 failure is the same as in https://github.com/numpy/numpy/pull/15791 and https://github.com/numpy/numpy/pull/15782.

**Comment by eric-wieser** [comment]
> Be more specific in an `except` statement: `object.__format__` raises
  TypeError when more than one argument is given.

What do you mean by "argument" here? Do you mean a non-empty format string?

**Comment by seberg** [comment]
@eric-wieser is that comment mis-placed or copied from somewhere else or am I just missing something obvious?

**Comment by eric-wieser** [comment]
Ah, that was unclear - I'm quoting the commit message

**Comment by WarrenWeckesser** [comment]
> What do you mean by "argument" here? Do you mean a non-empty format string?

Yes.  That commit is an update to https://github.com/numpy/numpy/pull/15410/commits/2c91824cdee50b8b50fc9b000195f4059631735e.  The `except` is changed to catch a more specific exception instead of `Exception` (as suggested by @seberg in  a comment there) but the actual exception raised is `TypeError`, not `ValueError`.

**Comment by eric-wieser** [comment]
I think what confused me is the message says _"more than one argument is given"_, but I think you meant "a non-empty argument is given".

Either way, all the commits can probably be squashed upon merge.

**Review by eric-wieser — APPROVED** [review]

File: numpy/ma/core.py [hunk 1]
```diff
@@ -6385,6 +6385,20 @@ def __repr__(self):
...
             return object.__repr__(self)
 
+    def __format__(self, format_spec):
+        # Replace ndarray.__format__ with the default, which supports no format characters.
+        # Supporting format characters is unwise here, because we do not know what type
+        # the user was expecting - better to not guess.
+        try:
+            return object.__format__(self, format_spec)
+        except TypeError:
+            warnings.warn(
```
Comment (eric-wieser) [inline numpy/ma/core.py]:
```suggestion
            # 2020-03-23, NumPy 1.19.0
            warnings.warn(
```

**Comment by WarrenWeckesser** [comment]
@eric-wieser, thanks for the reivew.   You're right, that wording in the commit isn't right.  But that whole commit message can be removed when the PR is squashed.

**Pull request merged**